frontend/state_management/Slices/AppViewSlice.ts

```diff
@@ -0,0 +1,54 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { AppViewState } from '../Reducers/AppViewReducer';
```

Comment:
Please just move the interface type from the old reducer into your new slice so we can delete old files without issues later.